### DISCUSSION TITLE
page.images

### DISCUSSION THREAD

**Discussion by yonglee7015** [py-pdf/pypdf#2135] [Q&A] [answered]
for image_file_object in page.images:
this always return all images in whole pdf, not the page we set.
is this a bug?

```
from pypdf import PdfReader

reader = PdfReader("example.pdf")

page = reader.pages[0]
count = 0

for image_file_object in page.images:
    with open(str(count) + image_file_object.name, "wb") as fp:
        fp.write(image_file_object.data)
        count += 1
```

**Comment by stefan6419846** [comment]
This might depend on the PDF file itself. Are you able to provide the one you have been testing this with?

**Reply by yonglee7015** [reply]
[AEO  1172.pdf](https://github.com/py-pdf/pypdf/files/12483807/AEO.1172.pdf)
Attached for your reference.

**Reply by yonglee7015** [reply]
Can you also test the page.extract_text() function? It seems always combine sentences in multiline without space.
the first page in my attached file.
![image](https://github.com/py-pdf/pypdf/assets/59268746/57e5bd26-e0dc-4556-8231-84902244ac7e)

**Reply by pubpub-zz** [reply]
This is a bug.
referenced in #2138

**Comment by pubpub-zz** [comment, marked as answer]
@yonglee7015 
your PDF is build in such a way that all images are attached to all the pages:
(displayed with pdfbox -debug)
![image](https://github.com/py-pdf/pypdf/assets/4083478/b634f948-3329-404b-b308-bd74a2d45918)

this is why you are getting all the images, not only the "selected for display" ones

**Reply by yonglee7015** [reply]
Thank you for your kindly help
